crates/forge_domain/src/temperature.rs

```diff
@@ -0,0 +1,215 @@
+use std::fmt;
+use std::ops::Deref;
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// A newtype for temperature values with built-in validation
+///
+/// Temperature controls the randomness in the model's output:
+/// - Lower values (e.g., 0.1) make responses more focused, deterministic, and
+///   coherent
+/// - Higher values (e.g., 0.8) make responses more creative, diverse, and
+///   exploratory
+/// - Valid range is 0.0 to 2.0
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd)]
+pub struct Temperature(f32);
+
+impl Temperature {
+    /// Creates a new Temperature value, returning an error if outside the valid
+    /// range (0.0 to 2.0)
+    pub fn new(value: f32) -> Result<Self, String> {
+        if Self::is_valid(value) {
+            Ok(Self(value))
+        } else {
+            Err(format!(
+                "temperature must be between 0.0 and 2.0, got {}",
+                value
+            ))
+        }
+    }
+
+    /// Creates a new Temperature value without validation
+    ///
+    /// # Safety
+    /// This function should only be used when the value is known to be valid
+    pub fn new_unchecked(value: f32) -> Self {
+        debug_assert!(Self::is_valid(value), "invalid temperature: {}", value);
+        Self(value)
+    }
+
+    /// Returns true if the temperature value is within the valid range (0.0 to
+    /// 2.0)
+    pub fn is_valid(value: f32) -> bool {
+        (0.0..=2.0).contains(&value)
+    }
+
+    /// Returns the inner f32 value
+    pub fn value(&self) -> f32 {
+        self.0
+    }
+}
+
+impl Deref for Temperature {
+    type Target = f32;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<Temperature> for f32 {
+    fn from(temp: Temperature) -> Self {
+        temp.0
+    }
+}
+
+impl fmt::Display for Temperature {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl Serialize for Temperature {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
```

Comment:
can we use serde_with for this https://docs.rs/serde_with/latest/serde_with/struct.DisplayFromStr.html